packages/react-native/src/private/webapis/performance/PerformanceObserver.js

```diff
@@ -253,33 +166,19 @@ export class PerformanceObserver {
       return;
     }
 
-    const observerConfig = registeredObservers.get(this);
-    if (!observerConfig) {
+    if (!this.#nativeObserverHandle) {
       return;
     }
 
-    // Disconnect this observer
-    for (const type of observerConfig.entryTypes.keys()) {
-      const numberOfObserversForThisType =
-        observerCountPerEntryType.get(type) ?? 0;
-      if (numberOfObserversForThisType === 1) {
-        observerCountPerEntryType.delete(type);
-        NativePerformanceObserver.stopReporting(
-          performanceEntryTypeToRaw(type),
-        );
-      } else if (numberOfObserversForThisType !== 0) {
-        observerCountPerEntryType.set(type, numberOfObserversForThisType - 1);
-      }
-    }
-
-    // Disconnect all observers if this was the last one
-    registeredObservers.delete(this);
-    if (registeredObservers.size === 0) {
-      NativePerformanceObserver.setOnPerformanceEntryCallback(undefined);
-      isOnPerformanceEntryCallbackSet = false;
-    }
+    NativePerformanceObserver.disconnect(this.#nativeObserverHandle);
+  }
 
-    applyDurationThresholds();
+  #createNativeObserver() {
+    return NativePerformanceObserver.createObserver(() => {
+      const entryList = new PerformanceObserverEntryList(NativePerformanceObserver.takeRecords(this.#observerHandle));
+      const droppedEntriesCount = NativePerformanceObserver.getDroppedEntriesCount(this.#nativeObserverHandle);
```

Comment:
We can improve performance slightly by avoiding this second call if this isn't the first time that the callback is called.